python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -768,63 +775,174 @@ def contains_properly(self, other, align=True):
-                {"idx": point_indices, "pip": point_result}
-            )
-            # if the number of points in the polygon is equal to the number of
-            # points, then the requirements for `.contains_properly` are met
-            # for this geometry type.
-            df_result = (
-                result.groupby("idx").sum().sort_index()
-                == result.groupby("idx").count().sort_index()
+            return ContainsProperlyBinpred(self, other, align)()
+
+    def equals(self, other, align=True):
+        """Compute if a GeoSeries of features A is equal to a GeoSeries of
+        features B. Features are equal if their coordinates are equal.
+
+        Parameters
+        ----------
+        other
+            a cuspatial.GeoSeries
+        align=True
+            to align the indices before computing .contains or not. If the
+            indices are not aligned, they will be compared based on their
+            implicit row order.
+
+        Examples
+        --------
+        Test if two points are equal:
+        >>> point = cuspatial.GeoSeries(
+            [Point(0, 0)],
             )
-            point_result = cudf.Series(
-                df_result["pip"], index=cudf.RangeIndex(0, len(df_result))
+        >>> point2 = cuspatial.GeoSeries(
+            [Point(0, 0)],
             )
-            point_result.name = None
-        return point_result
+        >>> print(point.equals(point2))
+        0    True
+        dtype: bool
+
+        Test if two polygons are equal:
+        >>> polygon = cuspatial.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            ]
+        )
+        >>> polygon2 = cuspatial.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            ]
+        )
+        >>> print(polygon.equals(polygon2))
+        0    True
+        dtype: bool
+
+        Returns
+        -------
+
+        result : cudf.Series
+            A Series of boolean values indicating whether each feature in A
+            is equal to the corresponding feature in B.
+        """
+        return EqualsBinpred(self, other, align)()
+
+    def covers(self, other, align=True):
+        """Compute if a GeoSeries of features A covers a second GeoSeries of
+        features B. A covers B if no points on B lie in the exterior of A.
+
+        Parameters
+        ----------
+        other
+            a cuspatial.GeoSeries
+        align=True
+            align the GeoSeries indexes before calling the binpred
+
+        Returns
+        -------
+        result : cudf.Series
+            A Series of boolean values indicating whether each feature in the
+            input GeoSeries covers the corresponding feature in the other
+            GeoSeries.
+        """
+        return CoversBinpred(self, other, align)()
+
+    def intersects(self, other, align=True):
+        """Compute the intersections of two GeoSeries.
```

Comment:
The documentation / definition here doesn't match the other predicates. This one says what it does but not what intersection means. Also, I think this computes *whether* they intersect, not the actual intersections.

Reply:
That was meant to be a short text, with a longer text beneath that I overlooked. I updated the text.